filament_scale_enhanced/__init__.py

```diff
@@ -46,19 +44,63 @@ def get_assets():
             less=["less/filament_scale.less"]
         )
 
+
     def on_startup(self, host, port):  # pylint: disable=unused-argument
-        self.hx = HX711(20, 21)
-        self.hx.set_reading_format("LSB", "MSB")
-        self.hx.reset()
-        self.hx.power_up()
-        self.t = octoprint.util.RepeatedTimer(3.0, self.check_weight)
-        self.t.start()
+        self.time = time.time()
+        
+        chip = self._settings.get(["chip"])
+        offset = float(self._settings.get(["offset"]))
+        cal_factor = float(self._settings.get(["cal_factor"]))
+
+        self._logger.info("Setting up scale, chip: [{}], offset: [{}], cal_factor: [{}]".format(chip, offset, cal_factor))
+
+        if chip == "nau7802":
+            from .scale.nau7802 import NAU7802
+
+            bus_id = int(self._settings.get(["nau_bus_id"]))
+
+            self._logger.debug("Initializing NAU7802, bus_id: [{}]".format(bus_id))
+            self.scale = NAU7802(offset, cal_factor, bus_id)
+            self._logger.debug("Initialized NAU7802, is_ready: [{}]".format(self.scale.is_ready()))
+        elif chip == "hx711":
+            from .scale.hx711 import HX711
+
+            clock_pin = self._settings.get(["hx_clockpin"])
+            data_pin = self._settings.get(["hx_datapin"])
+
+            self._logger.debug("Initializing HX711, clock_pin: [{}], data_pin: [{}]".format(clock_pin, data_pin))
+            self.scale = HX711(offset, cal_factor, clock_pin, data_pin)
+            self._logger.debug("Initialized HX711, is_ready: [{}]".format(self.scale.is_ready()))
+        else:
+            self.scale = None
```

Comment:
It'll save some boiler plate and give room for different backends in the future (maybe say an external arduino), if we `from . import scale` at the top of the file, then use getattr and pass the whole config payload into the library. Let each implementation handle parsing it during instantiation. Along with the logging statements. Potentially a try/except for if the backend configured doesn't exist

```suggestion
    self.scale = getattr(scale, chip)(settings=self._settings)
```

Reply:
That's a great idea and I'm all for simplifying this code.

- [ ] Move per-device instantiation details to `scale` library
- [ ] Implement config parsing in `scale` library to determine which subclass to init and handle per-subclass variables
- [ ] Use `getarr()` to init `self.scale` with the active config using the above implementation during `on_startup()`